src/pages/settings/integrations/AvalaraIntegrationMapItemDrawer/handleIntegrationMappingCUD.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  MappableTypeEnum,
+  MappingTypeEnum,
+  useCreateAnrokIntegrationCollectionMappingMutation,
+  useCreateAnrokIntegrationMappingMutation,
+  useDeleteAnrokIntegrationCollectionMappingMutation,
+  useDeleteAnrokIntegrationMappingMutation,
+  useUpdateAnrokIntegrationCollectionMappingMutation,
+  useUpdateAnrokIntegrationMappingMutation,
+} from '~/generated/graphql'
+import type {
+  BillingEntityForIntegrationMapping,
+  ItemMappingForMappable,
+  ItemMappingForNonTaxMapping,
+  ItemMappingForTaxMapping,
+} from '~/pages/settings/integrations/common'
+import { CreateUpdateDeleteSuccessAnswer } from '~/pages/settings/integrations/common/types'
+
+import type { FormValuesType } from './types'
+
+type CreateUpdateDeleteFunctions = {
+  createCollectionMapping: ReturnType<typeof useCreateAnrokIntegrationCollectionMappingMutation>[0]
+  createMapping: ReturnType<typeof useCreateAnrokIntegrationMappingMutation>[0]
+  deleteCollectionMapping: ReturnType<typeof useDeleteAnrokIntegrationCollectionMappingMutation>[0]
+  deleteMapping: ReturnType<typeof useDeleteAnrokIntegrationMappingMutation>[0]
+  updateCollectionMapping: ReturnType<typeof useUpdateAnrokIntegrationCollectionMappingMutation>[0]
+  updateMapping: ReturnType<typeof useUpdateAnrokIntegrationMappingMutation>[0]
+}
+
+export const handleIntegrationMappingCUD = async (
+  inputValues: FormValuesType['values'],
+  initialMapping:
+    | ItemMappingForTaxMapping
+    | ItemMappingForNonTaxMapping
+    | ItemMappingForMappable
+    | undefined,
+  formType: MappingTypeEnum | MappableTypeEnum,
+  integrationId: string,
+  {
+    createCollectionMapping,
+    createMapping,
+    deleteCollectionMapping,
+    deleteMapping,
+    updateCollectionMapping,
+    updateMapping,
+  }: CreateUpdateDeleteFunctions,
+  billingEntity: BillingEntityForIntegrationMapping,
+): Promise<CreateUpdateDeleteSuccessAnswer> => {
+  const isCollectionContext = (type: unknown): type is MappingTypeEnum => {
+    return Object.values(MappingTypeEnum).includes(type as MappingTypeEnum)
+  }
+
+  const hasItemValues = !!inputValues.externalId && !!inputValues.externalName
+
+  const isTaxContext =
+    !!initialMapping &&
+    'taxCode' in initialMapping &&
+    'taxNexus' in initialMapping &&
+    'taxType' in initialMapping
+  const hasInitialData = isTaxContext
+    ? !!initialMapping.taxCode && !!initialMapping.taxNexus && !!initialMapping.taxType
+    : !!initialMapping?.itemExternalId
+  const isCreate = !!initialMapping && !initialMapping.itemId && hasItemValues
+  const isEdit = !isCreate && hasInitialData && hasItemValues
+  const isDelete =
+    !isCreate && !isEdit && !hasItemValues && !!initialMapping && initialMapping.itemId
+
+  /**
+   * Happens since we launch this function for each billing entity, but some billing entities
+   * might not have any data to process (no initial data and no input data) = we want to keep the default mapping
+   */
+  if (!isCreate && !isEdit && !isDelete) {
```

Comment:
Find this condition cryptic and not clear when reading.

I understand it makes sense to rely on those variables as they contain useful check however I don't get why it would be a success if something is not a creation, edition nor deletion.

Would checking the presence of `initialMapping`, `hasInitialData` and `hasItemValues` be more explicit on the reason of this return presence?

Let's have a call to talk about it, might not grasp some edge cases reading file-per-file this PR